integration-tests/src/tests/client/sharding_upgrade.rs

```diff
@@ -963,37 +957,63 @@ fn setup_test_env_with_cross_contract_txs(
 
 // Test cross contract calls
 // This test case tests postponed receipts and delayed receipts
-#[test]
-fn test_shard_layout_upgrade_cross_contract_calls() {
+fn test_shard_layout_upgrade_cross_contract_calls_impl(resharding_type: ReshardingType) {
     init_test_logger();
 
     // setup
     let epoch_length = 5;
+    let genesis_protocol_version = get_genesis_protocol_version(&resharding_type);
+    let target_protocol_version = get_target_protocol_version(&resharding_type);
 
-    let (mut test_env, new_accounts) = setup_test_env_with_cross_contract_txs(epoch_length);
+    let (mut test_env, new_accounts) =
+        setup_test_env_with_cross_contract_txs(epoch_length, genesis_protocol_version);
 
     for _ in 1..5 * epoch_length {
-        test_env.step(0.);
+        test_env.step_impl(0., target_protocol_version, &resharding_type);
         test_env.check_receipt_id_to_shard_id();
     }
 
     let successful_txs = test_env.check_tx_outcomes(false, vec![2 * epoch_length + 1]);
-    let new_accounts: Vec<_> =
+    let new_accounts =
         successful_txs.iter().flat_map(|tx_hash| new_accounts.get(tx_hash)).collect();
+
     test_env.check_accounts(new_accounts);
 
     test_env.check_split_states_artifacts();
 }
 
+// Test cross contract calls
+// This test case tests postponed receipts and delayed receipts
+#[test]
+fn test_shard_layout_upgrade_cross_contract_calls_v1() {
+    test_shard_layout_upgrade_cross_contract_calls_impl(ReshardingType::V1);
+}
+
+// Test cross contract calls
+// This test case tests postponed receipts and delayed receipts
+#[cfg(feature = "protocol_feature_simple_nightshade_v2")]
+// TODO(resharding) this test is currently broken, potentially due to lack of
+// flat storage support. Once flat storage for resharding is fully implemented
+// this test should be revisited, fixed and re-enabled.
+#[ignore]
+#[test]
+fn test_shard_layout_upgrade_cross_contract_calls_v2() {
+    test_shard_layout_upgrade_cross_contract_calls_impl(ReshardingType::V2);
+}
```

Comment:
Is there an issue you could link here where readers can figure out or ask about the progress on fixing this?